### PR TITLE
[KEP-3085]: check PodReadyToStartContainers condition after gracefulshutdown

### PR DESCRIPTION
**Pull request by charles-chenzz**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
this PR check status of pod ready to start condition are set to false after terminating. 
#### Which issue(s) this PR fixes:

Fixes #

#### Special notes for your reviewer:
the test case should be add to promote kep 3085 to beta
#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/3085-pod-conditions-for-starting-completition-of-sandbox-creation#e2e-tests
```

### CODE REVIEW

File: test/e2e_node/node_shutdown_linux_test.go [hunk 1]
```diff
@@ -327,6 +328,28 @@ var _ = SIGDescribe("GracefulNodeShutdown [Serial] [NodeFeature:GracefulNodeShut
...
+			// All pod ready to start condition should set to false
+			gomega.Eventually(ctx, func(ctx context.Context) error {
+				list, err = e2epod.NewPodClient(f).List(ctx, metav1.ListOptions{
+					FieldSelector: nodeSelector,
+				})
+				if err != nil {
+					return err
+				}
+				framework.ExpectEqual(len(list.Items), len(pods), "the number of pods are not as expected")
+
```
Comment (charles-chenzz) [inline test/e2e_node/node_shutdown_linux_test.go:341]: this is why linter hints fails, should we replace? if so, I think the rest of the code should update as well
image

Comment (kannon92) [reply test/e2e_node/node_shutdown_linux_test.go:341]: You can replace your code.  And maybe open up a new PR switching over if you want for the others

Comment (charles-chenzz) [reply test/e2e_node/node_shutdown_linux_test.go]: updated

**Comment by kannon92** [comment]
https://storage.googleapis.com/k8s-triage/index.html?sig=node&test=GracefulNodeShutdown

This can be used to find the test that is running this.  You haven't triggered this test yet.

/test

**Comment by kannon92** [comment]
/retitle [KEP-3085]: check PodReadyToStartContainers condition after gracefulshutdown

**Review by SergeyKanzhelev — APPROVED** [review]
/lgtm
/approve

**Pull request merged**